src/core/shape_inference/include/utils.hpp

```diff
@@ -245,20 +245,20 @@ template <class TShape,
           class TRes = std::vector<TData>,
           class UnaryOperation = ov::util::Cast<TData>,
           typename std::enable_if<std::is_same<TShape, ov::PartialShape>::value>::type* = nullptr>
-std::unique_ptr<TRes> get_input_const_data_as(const ov::Node* op,
-                                              size_t idx,
-                                              const ITensorAccessor& tensor_accessor,
-                                              UnaryOperation&& func = ov::util::Cast<TData>()) {
+ov::optional<TRes> get_input_const_data_as(const ov::Node* op,
+                                           size_t idx,
+                                           const ITensorAccessor& tensor_accessor,
+                                           UnaryOperation&& func = ov::util::Cast<TData>()) {
     if (auto t = tensor_accessor(idx)) {
-        return std::unique_ptr<TRes>(new TRes(get_tensor_data_as<TData, TRes>(t, std::forward<UnaryOperation>(func))));
+        return ov::optional<TRes>(get_tensor_data_as<TData, TRes>(t, std::forward<UnaryOperation>(func)));
```

Comment:
What about style like in line 212 above.
```suggestion
        return {get_tensor_data_as<TData, TRes>(t, std::forward<UnaryOperation>(func))};
```

Reply:
This is under investigation as proposed suggestion can suppress RVO and maybe return to `ov::optional<>()` would be better option.

Reply:
Analysis show that in current implementation of `ov::optional` both options take advantage of RVO. I have decided to use shorter syntax.